### PR TITLE
PUB-510 - Rest Endpoints

### PR DESCRIPTION
**Pull request by ChrisS1512**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-510

### Change description ###

Separated out responsibilities of tests, added display names and remove uneeded test. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/rules/controllers/ControllerTest.java [hunk 1]
```diff
@@ -51,13 +51,43 @@ public void publicationThatExistsTest() throws Exception {
...
         List<CourtHearings> courtHearingsList = publication.getCourtHearingsList();
         assertThat(courtHearingsList.get(0).getCourtId()).as("Get first court id").isEqualTo(1);
+    }
+
+    @DisplayName("Should return a 200, with a publication that contains a multiple hearings")
+    @Test
+    public void publicationThatExistsHearingsTest() throws Exception {
+        MvcResult response = mockMvc.perform(get("/publication/1")).andExpect(status().isOk()).andReturn();
+
+        assertEquals(200, response.getResponse().getStatus(),
```
Comment (kurt-uk) [inline src/integrationTest/java/uk/gov/hmcts/reform/pip/rules/controllers/ControllerTest.java:78]: is this testing the same as line 76? in both youre only checkint the status, maybe return earlier on line 76 and do the assertion here or vice versa

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/rules/controllers/ControllerTest.java [hunk 2]
```diff
@@ -51,13 +51,43 @@ public void publicationThatExistsTest() throws Exception {
...
+
+        List<CourtHearings> courtHearingsList = publication.getCourtHearingsList();
+        List<Hearing> hearings = courtHearingsList.get(0).getHearingList();
 
-        List<Hearing> courtHearings = courtHearingsList.get(0).getHearingList();
-        assertThat(courtHearings.get(0).getHearingId()).as("Get first hearing id").isEqualTo(1);
-        assertThat(courtHearings.get(1).getHearingId()).as("Get second hearing id").isEqualTo(2);
+        assertThat(hearings.get(0).getHearingId()).as("Get first hearing id").isEqualTo(1);
+        assertThat(hearings.get(1).getHearingId()).as("Get second hearing id").isEqualTo(2);
     }
```
Comment (kurt-uk) [inline src/integrationTest/java/uk/gov/hmcts/reform/pip/rules/controllers/ControllerTest.java:91]: what happens if you do `/publication/2`?

Comment (kurt-uk) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/rules/controllers/ControllerTest.java:91]: or is that tested elsewhere?

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/rsecheck/controller/RootControllerTest.java [hunk 3]
```diff
@@ -33,7 +35,8 @@ public void testWelcomeMessage() {
     }
 
     @Test
-    public void testGetPublication() {
+    @DisplayName("Tests that a 200 code is returned, with a non null body")
+    public void testGetPublicationOkResponse() {
```
Comment (Matthew-Summerbell) [inline src/test/java/uk/gov/hmcts/reform/rsecheck/controller/RootControllerTest.java:39]: Negative test for if no publication is found?

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/rules/controllers/ControllerTest.java (hunk 2 above)
Comment (ChrisS1512) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/rules/controllers/ControllerTest.java:91]: /2 should return a 404.

Indeed. That's being covered on in the test on line 96

Comment (Matthew-Summerbell) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/rules/controllers/ControllerTest.java:91]: Yeah as Kurt states a not found test would be good

Comment (Matthew-Summerbell) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/rules/controllers/ControllerTest.java:91]: ahh yeah it was not showing up when expanded

File: src/integrationTest/java/uk/gov/hmcts/reform/pip/rules/controllers/ControllerTest.java (hunk 1 above)
Comment (ChrisS1512) [reply src/integrationTest/java/uk/gov/hmcts/reform/pip/rules/controllers/ControllerTest.java:78]: Good spot :) I've removed the checks via the mockMvc response, and done it via the assert instead.

I've updated the other tests in this file to do that too.

File: src/test/java/uk/gov/hmcts/reform/rsecheck/controller/RootControllerTest.java (hunk 3 above)
Comment (ChrisS1512) [reply src/test/java/uk/gov/hmcts/reform/rsecheck/controller/RootControllerTest.java:41]: Updated - added a test for this :)

**Review by Matthew-Summerbell — APPROVED** [review]

**Review by kurt-uk — APPROVED** [review]

**Pull request merged**